controller/src/main/java/io/pravega/controller/store/stream/ScaleMetadata.java

```diff
@@ -12,23 +12,27 @@
 import lombok.Data;
 
 import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
 
 @Data
+@NoArgsConstructor
+@AllArgsConstructor
 public class ScaleMetadata {
     /**
      * Time of scale operation.
      */
-    private final long timestamp;
+    private long timestamp;
```

Comment:
No. These need to be final, or this class will not be threadsafe.

Reply:
I see your point but it can't be a JAX-RS data structure as this. 

Maybe the real problem is that this class is exposed to the REST API, instead we should have another wrapper class to expose on it. A simple bean with getters and setters to be used as model for the API